cpp/src/arrow/compute/api_scalar.h

```diff
@@ -268,19 +268,43 @@ class ARROW_EXPORT ExtractRegexOptions : public FunctionOptions {
 /// Options for IsIn and IndexIn functions
 class ARROW_EXPORT SetLookupOptions : public FunctionOptions {
  public:
-  explicit SetLookupOptions(Datum value_set, bool skip_nulls = false);
+  enum NullMatchingBehavior { MATCH, SKIP, EMIT_NULL, INCONCLUSIVE };
+
+  explicit SetLookupOptions(Datum value_set, NullMatchingBehavior = MATCH);
+  // DEPRECATED(will be removed after removing of skip_nulls)
```

Comment:
Would `ARROW_DEPRECATED("will be removed after removing of skip_nulls")` better? Or this function would be removed later?

Reply:
Let's replace

Reply:
I misunderstood, these functions would be removed later indeed.

Reply:
Oh I got it, sorry for misleading